workbench-core/accounts/src/services/projectService.ts

```diff
@@ -254,6 +255,55 @@ export default class ProjectService {
     return newProject;
   }
 
+  /**
+   * Update the name or description of an existing project.
+   *
+   * @param request - a {@link UpdateProjectRequest} object that contains the id of the project to update
+   *                  as well as the new field values
+   * @returns a {@link Project} object that reflects the changes requested
+   */
+  public async updateProject(request: UpdateProjectRequest): Promise<Project> {
+    const { projectId, updatedValues } = request;
+
+    // disregard empty strings as valid input
+    if (updatedValues.name !== undefined && _.isEmpty(updatedValues.name)) {
+      updatedValues.name = undefined;
+    }
+    if (updatedValues.description !== undefined && _.isEmpty(updatedValues.description)) {
+      updatedValues.description = undefined;
+    }
+
+    // verify at least one attribute is being updated
+    if (!updatedValues.name && !updatedValues.description) {
+      throw Boom.badRequest('You must supply a new nonempty name and/or description to update the project.');
+    }
+
+    // if updating name, verify it is not in use and project exists
+    if (updatedValues.name) {
+      await Promise.all([this._isProjectNameInUse(updatedValues.name), this.getProject({ projectId })]);
+    } else {
+      // else, verify project still exists
+      await this.getProject({ projectId });
```

Comment:
I think we can refactor this so that we check first if the project exists, and then if it does, check if the name is being updated is valid. That way we don't need to worry about what might happen in an `else` clause in the future if changes get made.

Reply:
The design doc specified doing the checks in parallel. Your way works too but makes the code more sequential at the benefit of not having a condition in the code.